/*
 * This file was automatically generated by EvoSuite
 * Mon Jan 11 12:09:24 GMT 2021
 */

package com.iluwatar.saga.choreography;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import com.iluwatar.saga.choreography.ChoreographyChapter;
import com.iluwatar.saga.choreography.FlyBookingService;
import com.iluwatar.saga.choreography.ServiceDiscoveryService;
import java.util.NoSuchElementException;
import java.util.Optional;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class ServiceDiscoveryService_ESTest extends ServiceDiscoveryService_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      ServiceDiscoveryService serviceDiscoveryService0 = new ServiceDiscoveryService();
      FlyBookingService flyBookingService0 = new FlyBookingService(serviceDiscoveryService0);
      ServiceDiscoveryService serviceDiscoveryService1 = serviceDiscoveryService0.discover(flyBookingService0);
      ChoreographyChapter choreographyChapter0 = serviceDiscoveryService1.findAny();
      assertSame(flyBookingService0, choreographyChapter0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      ServiceDiscoveryService serviceDiscoveryService0 = new ServiceDiscoveryService();
      Optional<ChoreographyChapter> optional0 = serviceDiscoveryService0.find("");
      assertNotNull(optional0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      ServiceDiscoveryService serviceDiscoveryService0 = new ServiceDiscoveryService();
      // Undeclared exception!
      try { 
        serviceDiscoveryService0.findAny();
        fail("Expecting exception: NoSuchElementException");
      
      } catch(NoSuchElementException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("java.util.HashMap$HashIterator", e);
      }
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      ServiceDiscoveryService serviceDiscoveryService0 = new ServiceDiscoveryService();
      // Undeclared exception!
      try { 
        serviceDiscoveryService0.discover((ChoreographyChapter) null);
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // Cannot invoke \"com.iluwatar.saga.choreography.ChoreographyChapter.getName()\" because \"chapterService\" is null
         //
         verifyException("com.iluwatar.saga.choreography.ServiceDiscoveryService", e);
      }
  }
}
